odl/solvers/advanced/forward_backward.py

```diff
@@ -0,0 +1,168 @@
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+""" Implementation of the forward-backward primal-dual splitting algorithm for
+optimization.
+"""
+
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+from odl.operator import Operator
+
+
+__all__ = ('forward_backward_pd',)
+
+
+def forward_backward_pd(x, prox_f, prox_cc_g, L, grad_h, tau, sigma, niter,
+                        callback=None, **kwargs):
+    """ The forward-backward primal-dual splitting algorithm.
+
+    The algorithm minimizes the sum of several convex functionals composed with
+    linear operators,
+
+        ``min_x f(x) + sum_i g_i(L_i x) + h(x)``,
+
+    where ``f``, ``g_i`` are convex functionals, ``L_i`` are linear
+    `Operator`'s, and ``h`` is a convex and differentiable functional.
+
+    The method can also be used to solve the more general problem
+
+        ``min_x f(x) + sum_i (g_i @ l_i)(L_i x)``,
+
+    where ``l_i`` are convex functionals and @ is the infimal convolution:
+
+        ``(f @ g)(x) = inf_y { f(x-y) + g(y) }``
+
+    Parameters
+    ----------
+    x : `LinearSpaceVector`
+        Initial point, updated in place.
+    prox_f : `callable`
+        `Proximal factory` for the functional ``f``.
+    prox_cc_g : `sequence` of `callable`'s
+        Sequence of `proximal factories` for the convex conjuates of the
+        functionals ``g_i``.
+    grad_h : `Operator`
+        Operators representing the gradient of  `h`.
+    L : `sequence` of `Operator`'s
+        A sequence with as many elements as ``prox__cc_gs`` of operators
+        ``L_i``
+    tau : `float`
+        Step size-like parameter for ``prox_f``
+    sigma : `sequence` of  `float`
+        Sequence of step size-like parameter for the sequence ``prox_cc_g``
+    niter : `int`
+        Number of iterations
+    callback : `Callback`, optional
+        Show partial results
+
+    Other Parameters
+    ----------------
+    grad_cc_l : `sequence` of `Operator`'s, optional
+        Sequence of operators representing the gradient of  `l_i^*`.
+        If omitted, the simpler problem will be considered.
+
+    Notes
+    -----
+    The exact conditions on the involved functionals are as follows: :math:`f`
+    and :math:`g_i` are proper, convex and lower semicontinuous, and :math:`h`
+    is convex and differentialbe with :math:`\\eta`-Lipschitz continuous
+    gradient.
+
+    The optional input :math:`\\nabla l_i^*` need to be :math:`\\nu_i`
+    -Lipschitz continuous. Note that in the reference [BC2015]_, the condition
```

Comment:
I guess "-Lipschitz" will appear detached from the nu_i thing. Need to be on the same row.